src/org/zaproxy/zap/extension/history/PopupMenuPurgeHistory.java

```diff
@@ -97,7 +97,13 @@ private void purgeHistory(HistoryReference ref) {
 
         if (node.getHistoryReference() == ref) {
             // same active Node
-            PopupMenuPurgeSites.purge(map, node);
+            ExtensionHistory extHistory =
+                    Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class);
+
+            if (extHistory != null) {
+                extHistory.purge(map, node);
```

Comment:
Should use the instance variable "extension" instead (the null check can also be removed).